via/static/scripts/video_player/components/Transcript.tsx

```diff
@@ -27,43 +37,141 @@ export type TranscriptProps = {
    * the start of the video.
    */
   currentTime: number;
+
+  /**
+   * Callback invoked when the user selects a segment from the transcript.
+   */
+  onSelectSegment?: (segment: Segment) => void;
 };
 
 type TranscriptSegmentProps = {
   isCurrent: boolean;
+  onSelect: () => void;
   time: number;
   text: string;
 };
 
-function TranscriptSegment({ time, text }: TranscriptSegmentProps) {
-  // TODO - Use appropriate semantic elements here
+function TranscriptSegment({
+  isCurrent,
+  onSelect,
+  time,
+  text,
+}: TranscriptSegmentProps) {
   return (
-    <div>
-      <div>{time}</div>
-      <p>{text}</p>
-    </div>
+    <li
+      className={classnames(
+        'flex flex-row p-1 hover:text-black',
+        isCurrent && 'bg-grey-2',
+        !isCurrent && 'text-grey-6'
+      )}
+      data-is-current={isCurrent}
+      data-testid="segment"
+    >
+      <button className="pr-5 hover:underline" onClick={() => onSelect()}>
```

Comment:
Since the `onSelect` callback is not bind to an object, I think it should be fine to pass it as is, with no wrapping arrow function.

```suggestion
      <button className="pr-5 hover:underline" onClick={onSelect}>
```

Reply:
Hmm... in JavaScript, I would use the closure to explicitly discard the `Event` argument that get passed to `onClick`, and ensure upstream code doesn't depend on it. Now that we have TypeScript ensuring callers pass only a compatible callback, it is probably safe to skip the closure here.